src/builder.rs

```diff
@@ -222,7 +222,24 @@ impl Builder {
         let tmppath_macro = format!("_tmppath {}", self.rpmbuild_dir.join("tmp").display());
 
         // Calculate rpmbuild arguments
-        let args = ["-D", &topdir_macro, "-D", &tmppath_macro, "-ba", &spec_path];
+        let mut args = vec!["-D", &topdir_macro, "-D", &tmppath_macro, "-ba", &spec_path];
+
+        let arch = match self.config.metadata {
+            Some(crate::config::PackageMetadata {
+                rpm:
+                    Some(RpmConfig {
+                        target_architecture: Some(ref arch),
+                        ..
+                    }),
+                ..
+            }) => Some(arch.clone()),
+            _ => None,
+        };
```

Comment:
It seems like this could potentially be simplified via use of `and_then`. This code is probably bogus, but here's a rough suggestion:

```suggestion
        let arch = self.config.metadata.and_then(|metadata| metadata.rpm).and_then(|rpm| rpm.target_architecture);
```

Reply:
Was already wondering how it could be cleaner. Expect this during the weekend.